client/homebrew/editor/snippetbar/snippetbar.jsx

```diff
@@ -138,6 +149,35 @@ const Snippetbar = createClass({
 		});
 	},
 
+	replaceContent : function(item){
+		return this.props.updateBrew(item);
+	},
+
+	renderHistoryItems : function() {
+		const historyItems = getHistoryItems(this.props.brew);
+
+		return <div className='dropdown'>
+			{_.map(historyItems, (item, index)=>{
+				if(!item.savedAt) return;
```

Comment:
There should be a way we can hide the duplicate entries. I think it will be a source of confusion for a lot of people.

Reply:
I could change the functionality slightly: instead of using the current values instead of an empty slot, if the function instead skips the update for that slot, still cycling through the remaining slots. I expect this to result in the following functionality:
1. On first run, slot 1 gets filled (because it always gets filled with current data), so the menu would only show slot 1, version 1 data (current).
2. On second run, slot 2 is filled with slot 1's old data. Slots 3-5 remain unfilled. So slot 1, v2 data (current); slot 2, v1 data
3. Third run, slot 2 data progresses to 3, 1 to 2. So slot 1, v3 data (current); slot 2, v2 data; slot 3, v1 data.

...and so on until slot 1 has v5 data and slot 5 has v1 data.

As there will never be a point where multiple slots have the same data, it will (or at least, _should_) be clearer to the casual observer which slots contain the oldest data.

Reply:
It seems to be working perfectly now:

![image](https://github.com/user-attachments/assets/b96c056c-8439-414d-bc96-2790fc480bac)